wdl-ast/tests/validation/duplicate-decls/source.errors

```diff
@@ -0,0 +1,108 @@
+   ·
+19 │     Int x = y
+   │         ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `x`
+   ┌─ tests/validation/duplicate-decls/source.wdl:32:16
+   │
+26 │         Int x
+   │             - first declaration with this name is here
+   ·
+32 │         String x = "x"
+   │                ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `y`
+   ┌─ tests/validation/duplicate-decls/source.wdl:33:16
+   │
+27 │         Int y = 0
+   │             - first declaration with this name is here
+   ·
+33 │         String y = "y"
+   │                ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `x`
+   ┌─ tests/validation/duplicate-decls/source.wdl:38:9
+   │
+26 │         Int x
+   │             - first declaration with this name is here
+   ·
+38 │     Int x = y
+   │         ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `b`
+   ┌─ tests/validation/duplicate-decls/source.wdl:45:17
+   │
+28 │         String b
+   │                - first declaration with this name is here
+   ·
+45 │             Int b = 0
+   │                 ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `x2`
+   ┌─ tests/validation/duplicate-decls/source.wdl:46:17
+   │
+41 │         Int x2 = 0
+   │             -- first declaration with this name is here
+   ·
+46 │             Int x2 = 0
+   │                 ^^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `x`
+   ┌─ tests/validation/duplicate-decls/source.wdl:51:14
+   │
+26 │         Int x
+   │             - first declaration with this name is here
+   ·
+51 │     scatter (x in [1, 2, 3]) {
+   │              ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `z`
+   ┌─ tests/validation/duplicate-decls/source.wdl:52:13
+   │
+37 │     Int z = x
+   │         - first declaration with this name is here
+   ·
+52 │         Int z = x
+   │             ^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `nested`
+   ┌─ tests/validation/duplicate-decls/source.wdl:57:21
+   │
+55 │         scatter (nested in [1, 2, 3]) {
+   │                  ------ first declaration with this name is here
+56 │             scatter (baz in [1, 2, 3]) {
+57 │                 Int nested = 0
+   │                     ^^^^^^ this conflicts with a previous declaration of the same name
+
+error: conflicting declaration name `nested`
+   ┌─ tests/validation/duplicate-decls/source.wdl:64:13
+   │
+57 │                 Int nested = 0
+   │                     ------ first declaration with this name is here
+   ·
+64 │         Int nested = 0
+   │             ^^^^^^ this conflicts with a previous declaration of the same name
+
```

Comment:
I understand what's happening here, but it's a confusing situation. The `first declaration` with the name `nested` changes. Is there some way we can rephrase this so that these 2 errors back-to-back aren't contradicting each other?

Reply:
I can probably change the diagnostic for when it conflicts with a scatter variable to distinguish them.